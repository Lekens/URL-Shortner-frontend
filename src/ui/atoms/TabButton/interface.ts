export interface ITabButton {
  title: string;
  active?: boolean;
  extraClassName?: string;
  onClick: React.MouseEventHandler<HTMLButtonElement>;
  style: object;
}
