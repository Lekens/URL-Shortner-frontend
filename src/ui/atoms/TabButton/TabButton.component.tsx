import { useMemo } from 'react';

import cn from 'classnames';

import { ITabButton } from 'ui/atoms/TabButton/interface';

import styles from './styles.module.scss';

function TabButton({
  title,
  active,
  onClick,
  extraClassName,
  style,
  ...props
}: ITabButton) {
  const buttonStyle = useMemo(
    () =>
      cn(styles.button, extraClassName, {
        [styles.active]: active,
      }),
    [active, extraClassName],
  );

  return (
    <button
      style={style}
      type="button"
      className={buttonStyle}
      onClick={onClick}
      {...props}
    >
      <span className={styles.innerContent}>
        <span>{title}</span>
      </span>
    </button>
  );
}
export default TabButton;
