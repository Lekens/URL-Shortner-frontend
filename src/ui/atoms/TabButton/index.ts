export { default as TabButton } from './TabButton.component';
