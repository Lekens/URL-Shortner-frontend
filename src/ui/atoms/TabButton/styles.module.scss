@import 'features/common/assets/styles/variables.module.scss';

.button {
  padding: 18px 0;
  border: none;
  margin: 0 10px;
  background: inherit;
  color: $white;
  cursor: pointer;
  font-family: $font-bold;
  font-size: 14px;
  line-height: 18px;
  outline: none;
}

.active {
  padding-bottom: 9px;
  border-bottom: 4px solid $red;
  color: $red;
}

.innerContent {
  display: inline-block;
  vertical-align: middle;
}
