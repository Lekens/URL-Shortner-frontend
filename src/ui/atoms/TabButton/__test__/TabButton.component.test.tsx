import React from 'react';

import { shallow } from 'enzyme';

import TabButton from 'ui/atoms/TabButton/TabButton.component';
import { render } from '@testing-library/react';

describe('TabButton component', () => {
  it('Show match snapshot', () => {
    const executeClick = () => {};
    const { container } = render(
      <TabButton
        title="Hello Tab"
        onClick={executeClick}
        style={{ width: '20px' }}
      />,
    );
    expect(container).toMatchSnapshot();
  });

  it('Should find Hello Tab on Dom', () => {
    const executeClick = () => {};
    const container = shallow(
      <TabButton
        title="Hello Tab"
        onClick={executeClick}
        style={{ width: '20px' }}
      />,
    );

    expect(container.contains(<span>Hello Tab</span>)).toBe(true);
  });
});
