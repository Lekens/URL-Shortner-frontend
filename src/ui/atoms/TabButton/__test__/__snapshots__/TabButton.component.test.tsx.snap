// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabButton component Show match snapshot 1`] = `
<div>
  <button
    class="button"
    style="width: 20px;"
    type="button"
  >
    <span
      class="innerContent"
    >
      <span>
        Hello Tab
      </span>
    </span>
  </button>
</div>
`;
